### DISCUSSION TITLE
Testcontainer integration

### DISCUSSION THREAD

**Discussion by meetppr** [JanusGraph/janusgraph#4069] [Q&A] [answered]
Is there an integration of test containers for JanusGraph?
I need to use the test container for testing Junits against my Janusgraph service. Is it already available?

**Comment by porunov** [comment, marked as answer]
JanusGraph doesn't have Testcontainers image available. However, JanusGraph has Docker images which you can use in your Testcontainers. https://hub.docker.com/r/janusgraph/janusgraph
Moreover, you can use `janusgraph-inmemory` dependency in your Java application to quickly start JanusGraph server with inmemory storage: https://central.sonatype.com/artifact/org.janusgraph/janusgraph-inmemory
`inmemory` storage is quick and doesn't require you to setup any additional dependencies or any additional storage containers.

Starting inmemory storage is as easy as the following line. No other configurations are required:
```
graph = JanusGraphFactory.build().set('storage.backend', 'inmemory').open()
```

See more information about this type of storage here: https://docs.janusgraph.org/storage-backend/inmemorybackend/
